docs/proposals/20250124-From CAPD(docker) to CAPD(dev) .md

```diff
@@ -0,0 +1,162 @@
+
+`DevCluster` and `DevMachine` will have 
+- A `docker` backend, functionally equivalent to `DockerCluster` and `DockerMachine`
+- An `inMemory`, functionally equivalent to CAPIM's `InMemoryCluster` and `InMemoryMachine`
+- if and when Maintainers of CAPK(Kubemark) will choose to converge to CAPD(dev), a `kubemark` backend,  
+  functionally equivalent to `KubemarkMachine` (there is no `KubemarkCluster`)
+
+Below you can fin an example of `DockerCluster` and corresponding `DevMachine` using the docker backend:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: DockerMachine
+metadata:
+  name: controlplane
+spec:
+  extraMounts:
+    - containerPath: "/var/run/docker.sock"
+      hostPath: "/var/run/docker.sock"
+```
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: DevMachine
+metadata:
+  name: controlplane
+spec:
+  backend:
+    docker:
+      extraMounts:
+        - containerPath: "/var/run/docker.sock"
+          hostPath: "/var/run/docker.sock"
+```
+
+And an example of `InMemoryMachine` and corresponding `DevMachine` using the in memory backend:
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/alpha1
+kind: InMemoryMachine
+metadata:
+  name: in-memory-control-plane
+spec:
+  behaviour:
+    inMemory:
+      vm:
+        provisioning:
+          startupDuration: "10s"
+          startupJitter: "0.2"
+      node:
+        provisioning:
+          startupDuration: "2s"
+          startupJitter: "0.2"
+      apiServer:
+        provisioning:
+          startupDuration: "2s"
+          startupJitter: "0.2"
+      etcd:
+        provisioning:
+          startupDuration: "2s"
+          startupJitter: "0.2"
+```
+
+```yaml
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: DevMachine
+metadata:
+  name: in-memory-control-plane
+spec:
+  backend:
+    inMemory:
+      vm:
+        provisioning:
+          startupDuration: "10s"
+          startupJitter: "0.2"
+      node:
+        provisioning:
+          startupDuration: "2s"
+          startupJitter: "0.2"
+      apiServer:
+        provisioning:
+          startupDuration: "2s"
+          startupJitter: "0.2"
+      etcd:
+        provisioning:
+          startupDuration: "2s"
+          startupJitter: "0.2"
+```
+
+Please note that at the end of phase 1
+- `DockerCluster` and `DockerMachine` will continue to exist, thus allowing maintainers and users 
+  time to migrate progressively to the new kinds; however, it is important to notice that
+  `DockerCluster` and `DockerMachine` should be considered as deprecated from now on.
+- `InMemoryCluster` and `InMemoryMachine` will be removed, as well as the entire CAPIM provider.
+  (CAPIM is used only for CAPI scale tests, so users are not impacted and migration can be executed faster).
+
+Phase 2 consist of the actual removal of `DockerCluster` and `DockerMachine`; this phase will happen after maintainers 
```

Comment:
What happens to DockerMachinePool at this time? Will we keep it around and working with DevCluster and DevMachine?

(I assume lower overhead is to just also move it over to a DevMachinePool)

Reply:
> I assume lower overhead is to just also move it over to a DevMachinePool

I still hope someone steps up for MP, but I agree that move it over is the simplest move to do because it allows this proposal to move on no matter of the overall discuss about MP ownership